opentelemetry-api/tests/context/propagation/test_tracecontexthttptextformat.py

```diff
@@ -55,23 +58,25 @@ def test_headers_with_tracestate(self):
             span_id=format(self.SPAN_ID, "016x"),
         )
         tracestate_value = "foo=1,bar=2,baz=3"
-        span_context = FORMAT.extract(
-            get_as_list,
-            {
-                "traceparent": [traceparent_value],
-                "tracestate": [tracestate_value],
-            },
-        )
+        span_context = get_span_from_context(
+            FORMAT.extract(
+                get_as_list,
+                {
+                    "traceparent": [traceparent_value],
+                    "tracestate": [tracestate_value],
+                },
+            )
+        ).get_context()
         self.assertEqual(span_context.trace_id, self.TRACE_ID)
         self.assertEqual(span_context.span_id, self.SPAN_ID)
         self.assertEqual(
             span_context.trace_state, {"foo": "1", "bar": "2", "baz": "3"}
         )
-
-        mock_span = Mock()
-        mock_span.configure_mock(**{"get_context.return_value": span_context})
         output = {}  # type:typing.Dict[str, str]
-        FORMAT.inject(mock_span, dict.__setitem__, output)
+        span = trace.DefaultSpan(span_context)
```

Comment:
This surely works but it is unnecessarily coupling `DefaultSpan` to the test of `TraceContextFormat`. The `DefaultSpan` class should not be related to this test, we should be able to freely change the default implementation of `Span` without affecting this test. Even when the previous `Mock` and the current `DefaultSpan` object produce the same testing result, the former approach should be preferred because of the reason before.

Reply:
If the same mock is being used in several test cases, it can be placed in a common module and imported, as usual.

Reply:
I see the point about having some control over the implementation to note break unit tests randomly, but it looks as if this span is used very generally today.

I feel like adding a mock span is extra overhead and we'll be paying it by potentially needing updates to the DefaultSpan and MockSpan in many situations. Not strongly opinionated here but I personally like re-using code until we absolutely need to deviate.

Reply:
I don't feel strongly one way or another here, it's worth noting that the `TraceContextFormat` is already using the `DefaultSpan` though, so if I was using a MockSpan here, it could mean we end up having to make changes to both `DefaultSpan` and `MockSpan` when updating the `TraceContextFormat` in the future. If you feel strongly about this @ocelotl, I can change the test to use a `MockSpan`. Looks like we already have a couple of integrations that implement their own.

Reply:
Not feeling strongly either, if you find it more convenient that way, then go ahead, @codeboten :+1: .